src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -27,18 +27,19 @@ function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(Wrappe
         const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${props.policy?.id ?? '0'}`, {
             initWithStoredValues: false,
         });
+        const isConnectionDataFetchSkipped = isOffline || !props.policy || !props.policy.areConnectionsEnabled || !!hasConnectionsDataBeenFetched || !!props.policy.connections;
 
         useEffect(() => {
             // When the accounting feature is not enabled, or if the connections data already exists,
             // there is no need to fetch the connections data.
-            if (!props.policy || !props.policy.areConnectionsEnabled || !!hasConnectionsDataBeenFetched || !!props.policy.connections) {
+            if (isConnectionDataFetchSkipped || !props?.policy?.id) {
                 return;
             }
 
             openPolicyAccountingPage(props.policy.id);
-        }, [hasConnectionsDataBeenFetched, props.policy, isOffline]);
+        }, [hasConnectionsDataBeenFetched, props.policy, isOffline, isConnectionDataFetchSkipped]);
 
-        if (props.policy?.areConnectionsEnabled && (!props.policy || status === 'loading' || hasConnectionsDataBeenFetched === false)) {
+        if (props.policy?.areConnectionsEnabled && (!props.policy || status === 'loading')) {
```

Comment:
```suggestion
        if (props.policy?.areConnectionsEnabled && status === 'loading') {
```
The other condition will never be true since we check for props.policy?.areConnectionsEnabled before it